arangod/RestServer/DatabaseFeature.h

```diff
@@ -257,14 +249,39 @@ class DatabaseFeature : public ArangodFeature {
   std::unique_ptr<DatabaseManagerThread> _databaseManager;
   std::unique_ptr<IOHeartbeatThread> _ioHeartbeatThread;
 
-  std::atomic<DatabasesLists*> _databasesLists;
-  // TODO: Make this again a template once everybody has gcc >= 4.9.2
-  // arangodb::basics::DataProtector<64>
-  mutable basics::DataProtector _databasesProtector;
-  mutable Mutex _databasesMutex;
+  using DatabasesList = containers::FlatHashMap<std::string, TRI_vocbase_t*>;
```

Comment:
wouldn´t it be better to just use the Guarded data structure here?

Reply:
It's not same, we don't want write block reads

Reply:
Maybe it's possible, but it's big behavior difference

Reply:
More similar rcu approach, where data have epoch